src/qutip_qoc/_rl.py

```diff
@@ -220,14 +210,15 @@ def reset(self, seed=None):
         """
         self.save_episode_info()
 
-        time_diff = time.mktime(time.localtime()) - time.mktime(self._result.start_local_time)
+        time_diff = self.episode_info[-1]["elapsed_time"] - (self.episode_info[-2]["elapsed_time"] if len(self.episode_info) > 1 else time.mktime(self._result.start_local_time))
         self._result.iter_seconds.append(time_diff)
         self.current_step = 0                                           # Reset the step counter
         self.current_episode += 1                                       # Increment episode counter
         self.actions = self.temp_actions.copy()
         self.terminated = False
         self.truncated = False
         self.temp_actions = []
+        self._result._final_states = [self.state]
```

Comment:
In case the user wants to look for a shorter pulse duration:
Please make sure this is the correct final state, corresponding to the episode that has found the shortest duration (within all of the trial episodes).
`result` attributes like this could also be set within the result.py file.